staker/stateless_block_validator.go

```diff
@@ -567,21 +570,31 @@ func (v *StatelessBlockValidator) ValidateBlock(
 	if err != nil {
 		return false, err
 	}
-	if full {
-		gsEnd, err = v.validationSpawner.ExecuteArbitrator(ctx, input, moduleRoot)
+	var spawners []validator.ValidationSpawner
+	if useExec {
+		spawners = append(spawners, v.execSpawner)
 	} else {
-		gsEnd, err = v.validationSpawner.ExecuteJit(ctx, input, moduleRoot)
+		spawners = v.validationSpawners
 	}
-	if err != nil {
-		return false, err
+	if len(spawners) == 0 {
+		return false, errors.New("no validation defined")
 	}
-	return gsEnd == expEnd, nil
+	for _, spawner := range spawners {
+		gsEnd, err = spawner.Execute(ctx, input, moduleRoot)
+		if err != nil || gsEnd != expEnd {
+			return false, err
+		}
+	}
+	return true, nil
 }
 
 func (v *StatelessBlockValidator) RecordDBReferenceCount() int64 {
 	return v.recordingDatabase.ReferenceCount()
 }
 
 func (v *StatelessBlockValidator) Stop() {
-	v.validationSpawner.Stop()
+	v.execSpawner.Stop()
+	for _, spawner := range v.validationSpawners {
+		spawner.Stop()
```

Comment:
It's worth noting that this'll double-stop the execSpawner, but that's probably the safest behavior and we can just make sure that double-stopping spawners is fine.

Reply:
In this version execSpawner's Stop does nothing, and in the final version it's changed so the executionSpawner is not used as a validationSpawner (they are two separate connections to the same address).
Because we'll only merge everything together I think this can remain.